### ISSUE TITLE
Let users set on-chain fees for autopilot in settings

### ISSUE THREAD

**Issue by tanx** [lightninglabs/lightning-app#1298]

**Comment by otech47** [comment]
Exploring this a bit to see whether I should take it on.. I'm imagining a slider something like the one in Lightning Joule:

![Screen Shot 2019-08-23 at 11 37 09 AM](https://user-images.githubusercontent.com/4914611/63582973-5e586e80-c59a-11e9-95e4-e92f425e0308.png)

But there's some functional complexity there ^ with the "auto" and "slow" labels that may not be necessary. A simpler form could me a 1 sat minimum but the maximum would still need to be dynamically calculated from the fastest next block fee estimate I think. I don't think an arbitrary max would be a good idea.

Thoughts?

**Comment by otech47** [comment]
Much needed feature by the way! 👍 Would love to see this in there

**Comment by tanx** [comment]
@otech47 thanks for the feedback. We've discussed doing a slider that allow setting `sat/byte` but have decided to go with a high/medium/low prio setting in the payment ui. See: https://github.com/lightninglabs/lightning-app/issues/1296

Definitely open to a `sat/byte` slider for future iterations if we get feedback from users that they'd like more control. Thoughts @leegordo?

**Comment by bolatovumar** [comment]
@tanx if fee estimation api is goo then having high/medium/low toggle is fine I find. However, many wallets have subpar fee estimation and often even the "low" setting fees are too high. You are using LND's fee estimation (right?) which seems to be better than the average.

**Comment by otech47** [comment]
> @otech47 thanks for the feedback. We've discussed doing a slider that allow setting `sat/byte` but have decided to go with a high/medium/low prio setting in the payment ui. See: #1296
> 
> Definitely open to a `sat/byte` slider for future iterations if we get feedback from users that they'd like more control. Thoughts @leegordo?

Ok yeah I like the simplicity with that approach. I personally would like the ability to set 1 sat/byte so maybe the lowest toggle should be fixed to that, and medium/high can be estimated?

**Comment by bolatovumar** [comment]
@otech47 if 1 sat/byte is fixed and the user picks this option you could end up with a transaction that doesn't clear for a long time (or never) if there is a period of high demand for block space (in the next bull run or whatever).

**Comment by otech47** [comment]
> @otech47 if 1 sat/byte is fixed and the user picks this option you could end up with a transaction that doesn't clear for a long time (or never) if there is a period of high demand for block space (in the next bull run or whatever).

I agree this is possible, but could be acceptable for the near future. I've been using 1 sat/byte and have never waited more than a day maximum, it's almost always a matter of hours. But I suppose we would want to anticipate some level of congestion in the next 12-18 months... 

The Low setting definitely shouldn't be the default and maybe ~ 3-5 sats/byte would be more appropriate, just want to avoid the subpar fee estimation you mentioned.  Ideally a replace-by-fee mechanism could be triggered after X hours or something in the background...?

I built a script that analyzes recent blocks for low fee transactions and at this moment, it's showing 11% (2432 / 21175) of transactions in the last 10 blocks had a fee rate of  < 4 sats/byte

Maybe LND fee estimation is good enough though I haven't looked through that code

**Comment by tanx** [comment]
> > @otech47 if 1 sat/byte is fixed and the user picks this option you could end up with a transaction that doesn't clear for a long time (or never) if there is a period of high demand for block space (in the next bull run or whatever).
> 
> I agree this is possible, but could be acceptable for the near future. I've been using 1 sat/byte and have never waited more than a day maximum, it's almost always a matter of hours. But I suppose we would want to anticipate some level of congestion in the next 12-18 months...

That's a good point. I'm assuming we could add RBF in the tx details screen to mitigate that @Roasbeef? I like the idea of having a lower bound option for a final wallet sweeping tx when users likely don't really care when their tx is confirmed.

**Comment by bolatovumar** [comment]
@tanx i guess you could always have a fixed 1 sat/byte option and then "slow/medium/fast" options using the fee estimation API. This might be too complex from the UX perspective though.

**Comment by leegordo** [comment]
Yep me understand what the slider does?

**Comment by otech47** [comment]
> Yep me understand what the slider does?

@leegordo see: https://www.loom.com/share/26e93303d0bd4297a09b2caee9d6317c

**Comment by leegordo** [comment]
Got it! Thank you.
